src/uniq.js

```diff
@@ -18,4 +20,37 @@ var uniqWith = require('./uniqWith');
  *      R.uniq([1, '1']);     //=> [1, '1']
  *      R.uniq([[42], [42]]); //=> [[42]]
  */
-module.exports = uniqWith(equals);
+module.exports = (function() {
+  /**
+   * Uses a native `Set` instance where possible for
+   * removing duplicate items. Items that implement
+   * the fantasy-land Setoid spec fallback to using
+   * `_contains` to support custom equality behaviour.
+   */
+  function uniq(list) {
+    /* global Set */
+    var item,
+        set = new Set(),
+        idx = -1,
+        len = list.length,
+        items = [],
+        uniqs = [];
+
+    while (++idx < len) {
```

Comment:
Ramda's general attitude is that if you feed us sparse arrays where lists are expected, all bets are off.  Things may work fine, and you'll be happy with the results.  On the other hand, Ramda may corrupt your database, crash your car, steal your boyfriend, or insult your _latke_ recipe.  We take no responsibility whatsoever for its behavior if you can't be bothered to pass it a well-behaved list.

Ramda would really like to be dealing with lists, but since the languages doesn't support true lists, we make do with arrays as substitutes.  However,we simply assume that these are list-like dense arrays.